tools/celtest/test_runner.go

```diff
@@ -0,0 +1,707 @@
+
+// Package celtest provides functions for testing CEL policies and expressions.
+package celtest
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/interpreter"
+	"github.com/google/cel-go/test"
+	"github.com/google/cel-go/tools/compiler"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	celpb "cel.dev/expr"
+	conformancepb "cel.dev/expr/conformance/test"
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+	descpb "google.golang.org/protobuf/types/descriptorpb"
+	dynamicpb "google.golang.org/protobuf/types/dynamicpb"
+)
+
+var (
+	celExpression         string
+	testSuitePath         string
+	fileDescriptorSetPath string
+	configPath            string
+	baseConfigPath        string
+)
+
+func init() {
+	flag.StringVar(&testSuitePath, "test_suite_path", "", "path to a test suite")
+	flag.StringVar(&fileDescriptorSetPath, "file_descriptor_set", "", "path to a file descriptor set")
+	flag.StringVar(&configPath, "config_path", "", "path to a config file")
+	flag.StringVar(&baseConfigPath, "base_config_path", "", "path to a base config file")
+	flag.StringVar(&celExpression, "cel_expr", "", "CEL expression to test")
+	flag.Parse()
+}
+
+// TestRunnerOption is used to configure the following attributes of the Test Runner:
+// - set the Compiler
+// - add Input Expressions
+// - set the test suite file path
+// - set the test suite parser based on the file format: YAML or Textproto
+type TestRunnerOption func(*TestRunner) (*TestRunner, error)
+
+// TriggerTests triggers tests for a CEL policy, expression or checked expression
+// with the provided set of options. The options can be used to:
+// - configure the Compiler used for parsing and compiling the expression
+// - configure the Test Runner used for parsing and executing the tests
+func TriggerTests(t *testing.T, testRunnerOpts []TestRunnerOption, testCompilerOpts ...any) {
+	testRunnerOptions := testRunnerOptions(testRunnerOpts, testCompilerOpts...)
+	tr, err := NewTestRunner(testRunnerOptions...)
+	if err != nil {
+		t.Fatalf("error creating test runner: %v", err)
+	}
+	programs, err := tr.Programs(t)
+	if err != nil {
+		t.Fatalf("error creating programs: %v", err)
+	}
+	tests, err := tr.Tests(t)
+	if err != nil {
+		t.Fatalf("error creating tests: %v", err)
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := tr.ExecuteTest(t, programs, test)
+			if err != nil {
+				t.Fatalf("error executing test: %v", err)
+			}
+		})
+	}
+}
+
+func testRunnerOptions(testRunnerOpts []TestRunnerOption, testCompilerOpts ...any) []TestRunnerOption {
```

Comment:
Could you actually make this a functional option to configure common options from flags? Can be done in a follow-up

```go
func TestOptionsFromFlags() TestRunnerOption {
   return func(tr *TestRunner) (*TestRunner, error) {
      opts := []TestRunnerOption{
          DefaultTestSuiteParser(testSuitePath),
          AddFileDescriptorSet(...),
          ...
       }
       var err error
       for _, o := range opts {
         tr, err = o(tr)
         if err != nil {
           return nil, err
         }
       }
       return tr
   }
}

Reply:
Resolving for now, in order to merge, but please update in the next PR